Code/Tools/ProjectManager/Source/CreateAGemScreen.cpp

```diff
@@ -406,81 +409,156 @@ namespace O3DE::ProjectManager
-            bool gemPathValid = ValidateGemPath();
-            if (gemNameValid && gemDisplayNameValid && licenseValid && gemPathValid)
-            {
-                m_createGemInfo.m_displayName = m_gemDisplayName->lineEdit()->text();
-                m_createGemInfo.m_name = m_gemName->lineEdit()->text();
-                m_createGemInfo.m_summary = m_gemSummary->lineEdit()->text();
-                m_createGemInfo.m_requirement = m_requirements->lineEdit()->text();
-                m_createGemInfo.m_licenseText = m_license->lineEdit()->text();
-                m_createGemInfo.m_licenseLink = m_licenseURL->lineEdit()->text();
-                m_createGemInfo.m_documentationLink = m_documentationURL->lineEdit()->text();
-                m_createGemInfo.m_path = m_gemLocation->lineEdit()->text();
-                m_createGemInfo.m_features = m_userDefinedGemTags->getTags();
-
-                m_stackWidget->setCurrentIndex(GemCreatorDetailsScreen);
-                m_nextButton->setText(tr("Create"));
-                m_gemCreatorDetailsTab->setEnabled(true);
-                m_gemCreatorDetailsTab->setChecked(true);
-            }
+            ProceedToGemCreatorDetailsPage();
         }
         else if (m_stackWidget->currentIndex() == GemCreatorDetailsScreen)
         {
-            bool originIsValid = ValidateFormNotEmpty(m_origin);
-            bool repoURLIsValid = ValidateRepositoryURL();
-            if (originIsValid && repoURLIsValid)
-            {
-                m_createGemInfo.m_origin = m_origin->lineEdit()->text();
-                m_createGemInfo.m_originURL = m_originURL->lineEdit()->text();
-                m_createGemInfo.m_repoUri = m_repositoryURL->lineEdit()->text();
-                QString templateLocation;
-                if (m_formFolderRadioButton->isChecked())
-                {
-                    templateLocation = m_gemTemplateLocation->lineEdit()->text();
-                }
-                else
-                {
-                    int templateID = m_radioButtonGroup->checkedId();
-                    templateLocation = m_gemTemplates[templateID].m_path;
-                }
-                auto result = PythonBindingsInterface::Get()->CreateGem(templateLocation, m_createGemInfo);
-                if (result.IsSuccess())
-                {
-                    emit GemCreated(result.GetValue<GemInfo>());
-                    emit GoToPreviousScreenRequest();
-                }
-                else
-                {
-                    QMessageBox::critical(
-                        this,
-                        tr("Failed to create gem"),
-                        tr("The gem failed to be created"));
-                }
-            }
+            ProceedToGemAction();
         }
     }
 
+    //this ignores the templates page
+    void CreateGem::ClearFields()
+    {
+        //details page
+        m_gemDisplayName->lineEdit()->clear();
```

Comment:
hmm wouldn't it be nice if you could just issue a single line "reset" signal to any fields in a form listening for it and they would clear/reset error labels?
e.g. our field widgets would connect to the parent (or maybe some passed in QObject?) listening for the signal
```
connect(parent, SIGNAL(reset()), this, SLOT(reset()));
```
and forms/screens could call it with
```
emit reset();
```
A possible future handy bit of functionality

Reply:
I'll mark this as a github issue so that it can be addressed in the future.